src/workbench/P5VisualizerTemplate.ts

```diff
@@ -0,0 +1,134 @@
+import {P5Visualizer} from '../visualizers/P5Visualizer'
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+
+/** md
+# p5 Visualizer Template
+
+[image should go here]
+
+This template can be used as a starting point for designing a p5 visualizer.
+
+## Parameters
+**/
+
+class P5VisualizerTemplate extends P5Visualizer {
+    name = 'p5 Visualizer Template' // the name shown on the visualizer menu
+
+    // external parameters, which the user can choose while creating the
+    // visualizer bundle
+    baseIndex = 0 // the default value of this parameter is zero
+    params = {
+        /** md
+- baseIndex: Where in the sequence to start browsing.
+         **/
+        baseIndex: {
+            value: this.baseIndex,
+            displayName: 'Base index',
+            required: true,
+        },
+    }
+
+    // internal parameters. the value of `index` will always be overwritten in
+    // `setup`, but TypeScript can't infer that, so we have to give `index` an
+    // initial value
+    index = 0
+    flash = 0
+
+    checkParameters() {
+        const status = super.checkParameters()
+
+        // make sure the base index doesn't precede the first index
+        if (this.params.baseIndex.value < this.seq.first) {
+            status.isValid = false
+            status.errors.push(
+                "The base index can't precede the"
+                    + ` first index, ${this.seq.first}`
+            )
+        }
+
+        return status
+    }
+
+    setup() {
+        // run the setup process for a general p5 visualizer, and make sure the
+        // p5 sketch has been successfully created. most p5 visualizers should
+        // do this
```

Comment:
I think you need to strengthen that statement. Any P5 visualizer must do this unless they have really good reason to take over canvas creation themselves, and then they have to make sure to perform all of the actions of the base class setup() themselves.

Reply:
The comment and visualizer-building guide now both have an unqualified instruction to call `super.setup()`. They also explain that `super.setup()` includes the `createCanvas()` call required by p5.